### PR TITLE
gui/renderer: Shader debugging utilities

### PR DESCRIPTION
**Pull request by VelocityRa**
image

### CODE REVIEW

File: src/emulator/gui/include/gui/functions.h [hunk 1]
```diff
@@ -42,3 +43,17 @@ void draw_game_selector(HostState &host);
 void draw_reinstall_dialog(HostState &host, GenericDialogState *status);
 
 } // namespace gui
+
+// Extensions to ImGUi
+namespace ImGui {
+
+static auto vector_getter = [](void *vec, int idx, const char **out_text) {
```
Comment (pent0) [inline src/emulator/gui/include/gui/functions.h]: What's the point of this being a lambda.

Comment (VelocityRa) [reply src/emulator/gui/include/gui/functions.h]: oh yea it was originally inside a func, forgot to make it a normal func when it was moved out

File: src/emulator/gui/src/shader_editor_dialog.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+#include <stdexcept>
+
+#define EXTERNAL_DISASM_ENABLE 0
+#define EXTERNAL_DISASM_PATH ""
+#define EXTERNAL_DISASM_ARGS ""
+
+namespace util {
+
+template <size_t MaxOutSize>
+std::string exec(std::string cmd) {
```
Comment (pent0) [inline src/emulator/gui/src/shader_editor_dialog.cpp]:
```suggestion
std::string exec(const std::string &cmd) {
```

I want to put this in utils. Don't want cross-platform code hanging around in gui.

Comment (pent0) [reply src/emulator/gui/src/shader_editor_dialog.cpp]: You can still put this here, im just making suggestion

Comment (VelocityRa) [reply src/emulator/gui/src/shader_editor_dialog.cpp]: yea fair enough was just being lazy tbh

**Pull request merged**